File: Prova_1/logaritmo.m
function img = logaritmo(img, c)

    img = im2double(img);

    [rows, columns] = size(img);

    for i = 1: rows

        for j = 1: columns

            img(i, j) = c * log( 1 + img(i, j));

        end
    end
end

File: Prova_1/questao_2_python/transforms/logaritmo.py
from numpy import log
from skimage.util import img_as_float


def logaritmo(img, c):
    img = img_as_float(img)

    rows, columns = img.shape

    for i in range(1, rows):
        for j in range(1, columns):
            img[i, j] = c * log(1 + img[i, j])

    return img
